workflow.rst

```diff
@@ -813,6 +813,98 @@ place::
         }
     }
 
+Creating Your Own Marking Store
+-------------------------------
+
+You may need to implement your own store to execute some additional logic
+when the marking is updated. For example, you may have some specific needs
+to store the marking on certain workflows. To do this, you need to implement
+the
+:class:`Symfony\\Component\\Workflow\\MarkingStore\\MarkingStoreInterface`::
+
+    namespace App\Workflow\MarkingStore;
+
+    use Symfony\Component\Workflow\Marking;
+    use Symfony\Component\Workflow\MarkingStore\MarkingStoreInterface;
+
+    class ReflectionMarkingStore implements MarkingStoreInterface
+    {
+        /**
+         * @var string
+         */
+        private $property;
+
+        public function __construct(string $property = 'property')
+        {
+            $this->property = $property;
+        }
+
+        public function getMarking(object $subject): Marking
+        {
+            $r = new \ReflectionObject($subject);
+            $property = $r->getProperty($this->property);
+
+            return new Marking([$property->getValue($subject) => 1]);
+        }
+
+        public function setMarking(object $subject, Marking $marking): void
+        {
+            $r = new \ReflectionObject($subject);
+            $property = $r->getProperty($this->property);
+
+            $property->setValue($subject, $marking);
+        }
+    }
```

Comment:
I would simplify this a lot
* Don't use the reflection
* hardcode the property
* typehint with this real object

Since this is an example, we don't need to abstract things

Reply:
Good idea! That's updated :slightly_smiling_face: